### PR TITLE
Add option to deploy to Docker Hub

### PR DESCRIPTION
**Pull request by fbricon**
Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,90 +1,81 @@
...
-          }
-          setDeployResponse(output);
-        });
+  async function waitAndOpen(route: string, currentOutput: string) {
+    let waitRoute = route;
+    if (process.env.NODE_ENV === 'development') {
+      waitRoute = `${process.env.REACT_APP_CORS_PROXY_URL}/${route}`;
+    }
+    let output = currentOutput;
+    await waitOnUrl(`${waitRoute}`, WAITING_ON_URL_TIMEOUT, 1000, (out) => {
```
Comment (dgolovin) [inline client/src/App.tsx:62]: `${waitRoute}` => waitRoute

Comment (dgolovin) [inline client/src/App.tsx:62]: `${waitRoute}` should be just waitRoute.

File: client/src/components/deployButton.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    setSelectedIndex(index);
+    setOpen(false);
+  };
+
+  const handleToggle = () => {
+    setOpen((prevOpen) => !prevOpen);
+  };
+
+  const handleClose = (event: Event) => {
+    if (
```
Comment (dgolovin) [inline client/src/components/deployButton.tsx:43]:
``` javascript
setOpen( !!anchorRef.current && anchorRef.current.contains(event.target as HTMLElement))
```
or

``` javascript
setOpen(anchorRef.current !== null && anchorRef.current.contains(event.target as HTMLElement))
```
wold look better

**Review by dgolovin — APPROVED** [review]
LGTM

File: client/src/components/deployButton.tsx (hunk 2 above)
Comment (fbricon) [reply client/src/components/deployButton.tsx:43]: no it's a different logic. basically if (condition is met) then don't change the open state, else set open state to false.

Anyways, this code is coming from https://mui.com/material-ui/react-button-group/#split-button, so I trust it more than anything I can write ;-)

File: client/src/App.tsx (hunk 1 above)
Comment (fbricon) [reply client/src/App.tsx]: fixed

**Pull request merged**